packages/discord.js/src/structures/Emoji.js

```diff
@@ -46,7 +46,7 @@ class Emoji extends Base {
    * @returns {?string}
    */
   imageURL(options) {
-    return this.id && this.client.rest.cdn.emoji(this.id, options);
+    return this.id && this.client.rest.cdn.emoji(this.id, this.animated ? { ...options, extension: 'gif' } : options);
```

Comment:
This change makes it so when an emoji is animated, passing `extension: 'png'` will still return a GIF.

However, if you swap the order, `...options` will be able to override the `extension` property:
```suggestion
    return this.id && this.client.rest.cdn.emoji(this.id, this.animated ? { extension: 'gif', ...options } : options);
```

Reply:
I would also document this dynamic default so users know what to expect.